Basic JSON API definitions

PetriHaikonen opened the pull request:
- Add basic JSON API definitions
- GitHub actions for automating releases and tags
- IntelliJ IDE configuration files added to .gitignore

Commit:
GitHub actions for auto-tagging from pull requests

Commit:
Basic JSON API definitions

Commit:
Add IntelliJ IDE configuration files to .gitignore

tlangens commented on resources/JsonApi.yaml:
```diff
@@ -0,0 +1,276 @@
+  ####################
+  # Common responses
+  ####################
+  responses:
```
There's actually a couple more in [Kinescope](https://github.com/24i/prd-kinescope/blob/master/resources/openapi/JsonApi.yaml)
Let's add those too.

PetriHaikonen replied:
I also added a few additional ones from importer project.

Commit:
Added following common responses to basic JSON API definitions

- 403 Forbidden
- 409 Conflict
- 413 Payload Too Large
- 500 Internal Server Error
- 503 Service Unavailable

tlangens commented on resources/JsonApi.yaml:
```diff
@@ -268,6 +287,81 @@ components:
+                        code: '10100'
+                        title: 'Payload Too Large'
+                        detail: 'Request payload is too large'
+    500:
+      description: Internal Server Error
+      content:
+        application/vnd.api+json:
+          schema:
+            type: object
+            required:
+              - errors
+            properties:
+              errors:
+                type: array
+                items:
+                  allOf:
+                    - $ref: '#/components/schemas/Error'
+                    - example:
+                        status: '500'
+                        title: 'Internal Server Error'
+                        detail: 'An error occurred while processing the request'
+    503:
+      description: Service Unavailable
+      content:
+        application/vnd.api+json:
+          schema:
+            type: object
+            required:
+              - errors
+            properties:
+              errors:
+                type: array
+                items:
+                  allOf:
+                    - $ref: '#/components/schemas/Error'
+                    - example:
+                        status: '503'
+                        code: '10000'
+                        title: 'Service Unavailable'
+                        detail: 'Service is currently not available'
```
No blocker, but we wouldn't typically list these in the API definitions because they are never _expected_ to be returned.
